src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -422,9 +422,32 @@ internal Accessibility LocalAccessibility
         /// <summary>
         /// Indicates whether this accessor is readonly due to reasons scoped to itself and its containing property.
         /// </summary>
-        internal override bool IsDeclaredReadOnly => LocalDeclaredReadOnly || _property.HasReadOnlyModifier || IsReadOnlyAutoGetter;
+        internal override bool IsDeclaredReadOnly
+        {
+            get
+            {
+                if (LocalDeclaredReadOnly || _property.HasReadOnlyModifier)
+                {
+                    return true;
+                }
+
+                // If we have IsReadOnly..ctor, we can use the attribute. Otherwise, we need to NOT be a netmodule and the type must not already exist in order to synthesize it.
+                var isReadOnlyAttributeUsable = DeclaringCompilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_IsReadOnlyAttribute__ctor) != null ||
```

Comment:
>DeclaringCompilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_IsReadOnlyAttribute__ctor) [](start = 48, length = 118)

It seems like the existence of the attribute should only affect emit, not whether this accessor reports that it is `IsDeclaredReadOnly`. Or perhaps I am misunderstanding how this property is used.

Reply:
Consider that if we did that C# would begin treating this type differently when used within the compilation unit and when used from an external one. That seems wrong. Also it means the public API of the type will differ observably when it's in source vs. re-hydrated from metadata.